powerauth-backend-tests/src/test/java/com/wultra/security/powerauth/test/v31/PowerAuthIdentityVerificationTest.java

```diff
@@ -1057,14 +1124,17 @@ private void verifyOtpCheck(String processId) throws Exception {
                 assertTrue(stepLogger.getResult().isSuccess());
                 assertEquals(200, stepLogger.getResponse().getStatusCode());
 
-                for (StepItem item : stepLogger.getItems()) {
-                    if (item.getName().equals("Decrypted Response")) {
-                        String responseData = item.getObject().toString();
-                        ObjectResponse<OtpVerifyResponse> objectResponse = objectMapper.readValue(responseData, new TypeReference<ObjectResponse<OtpVerifyResponse>>() {});
-                        OtpVerifyResponse response = objectResponse.getResponseObject();
-                        otpVerified = response.isVerified();
-                    }
-                }
+                otpVerified = stepLogger.getItems().stream()
+                        .filter(isStepItemDecryptedResponse())
+                        .map(StepItem::getObject)
+                        .map(Object::toString)
+                        .map(it -> safeReadValue(it, new TypeReference<ObjectResponse<OtpVerifyResponse>>() {}))
+                        .filter(Objects::nonNull)
+                        .map(ObjectResponse::getResponseObject)
+                        .map(OtpVerifyResponse::isVerified)
+                        .findFirst()
```

Comment:
Stream and its `#findFirst` makes it _short-circuit evaluation_, or it may be extracted to a method.